### PR TITLE
feat: spam protection for create/update referral program

### LINKED ISSUE #10563: [Feature]: Require a party creating/updating referral program to have min balance
Require a party creating/updating referral program to have min balance - see https://github.com/vegaprotocol/specs/issues/2154. If the party has insufficient balance - reject the transaction on check tx.

### PR DESCRIPTION
**Pull request by ze97286**
Closes #10563

### CODE REVIEW

**Review by jeremyletang — APPROVED** [review]

File: core/netparams/keys.go [hunk 1]
```diff
@@ -173,6 +173,7 @@ const (
 	SpamProtectionMaxApplyReferralCode     = "spam.protection.max.applyReferralCode"
 	SpamProtectionBalanceSnapshotFrequency = "spam.protection.balanceSnapshotFrequency"
 	SpamProtectionApplyReferralMinFunds    = "spam.protection.applyReferral.min.funds"
+	SpamProtectionReferralProgramMinFunds  = "spam.protection.referralProgram.min.funds"
```
Comment (davidsiska-vega) [inline core/netparams/keys.go:176]: It probably wants to be `minFunds` but given the parameter above (line 175) it's probably too late for that.

**Review by davidsiska-vega — CHANGES_REQUESTED** [review]
Can we just confirm we're doing the quantum conversion on balances somewhere?

File: core/referral/engine.go [hunk 2]
```diff
@@ -115,6 +119,18 @@ func (e *Engine) CheckSufficientBalanceForApplyReferralCode(party types.PartyID,
 	return nil
 }
 
+func (e *Engine) CheckSufficientBalanceForCreateOrUpdateReferralProgram(party types.PartyID, balance *num.Uint) error {
+	if balance.LT(e.minBalanceForReferralProgram) {
```
[the remark this answers is not in the document]
Comment (ze97286) [reply core/referral/engine.go:123]: balance (from getPartyBalance) is already quantum converted, min funds is in qunatum units.

File: core/netparams/keys.go (hunk 1 above)
Comment (ze97286) [reply core/netparams/keys.go:176]: yeah.

**Comment by ze97286** [comment]
> Can we just confirm we're doing the quantum conversion on balances somewhere?

https://github.com/vegaprotocol/vega/blob/develop/core/collateral/engine.go#L213
https://github.com/vegaprotocol/vega/blob/develop/core/collateral/engine.go#L173

**Review by davidsiska-vega — APPROVED** [review]

**Comment by davidsiska-vega** [comment]
> > Can we just confirm we're doing the quantum conversion on balances somewhere?
> 
> https://github.com/vegaprotocol/vega/blob/develop/core/collateral/engine.go#L213 https://github.com/vegaprotocol/vega/blob/develop/core/collateral/engine.go#L173

Thank you.

**Pull request merged**